routing/index_graph.hpp

```diff
@@ -76,13 +207,168 @@ class IndexGraph final
+
+  /// \brief Calls |f| for |restrictionPoint| and for all restriction points formed by
+  /// all ingoing and outgoing edges which were generated from the ingoing edge
+  /// and the outgoing edge of |restrictionPoint|.
+  void ApplyRestrictionRealFeatures(RestrictionPoint const & restrictionPoint,
+                                    function<void(RestrictionInfo const &)> && f);
+
+  /// \brief Fills |path| with points from point |from| to point |to|
+  /// \note |from| and |to| should belong to the same feature.
+  /// \note The order on points in items of |connectionPaths| is from |from| to |to|.
+  void GetSingleFeaturePath(RoadPoint const & from, RoadPoint const & to,
+                            vector<RoadPoint> & path);
+
+  /// \brief Fills |path| with a path from |from| to |to| of points of |featureId|.
+  void GetFeatureConnectionPath(Joint::Id from, Joint::Id to, uint32_t featureId,
+                                vector<RoadPoint> & path);
+
+  void GetOutgoingGeomEdges(vector<JointEdge> const & outgoingEdges, Joint::Id center,
+                            vector<JointEdgeGeom> & outgoingGeomEdges);
+
+  /// \brief Fills |connectionPaths| with all path from joint |from| to joint |to|.
+  /// If |from| and |to| don't belong to the same feature |connectionPaths| will
+  /// be empty.
+  /// If |from| and |to| belong to only one feature |connectionPaths| will have one item.
+  /// It's the most common case.
+  /// If |from| and |to| could be connected by several features |connectionPaths|
+  /// will have several items.
+  /// \note The order on points in items of |connectionPaths| is from |from| to |to|.
+  void GetConnectionPaths(Joint::Id from, Joint::Id to,
+                          vector<vector<RoadPoint>> & connectionPaths);
+
+  void CreateFakeFeatureGeometry(vector<RoadPoint> const & geometrySource, double speed,
+                                 RoadGeometry & geometry);
+
+  void GetNeighboringEdge(RoadGeometry const & road, RoadPoint const & rp, bool forward, bool outgoing,
+                          bool graphWithoutRestrictions, vector<JointEdge> & edges) const;
+
+  double GetSpeed(RoadPoint const & rp);
+
+  /// \brief Finds neghboring of |centerId| joint on feature id of |center|
+  /// which is contained in |edges| and fills |oneStepAside| with it.
+  /// \note If oneStepAside is empty no neghboring nodes were found.
+  /// \note Taking into account the way of setting restrictions almost always |oneStepAside|
+  /// will contain one or zero items. Meanwhile the it it's posible to draw map
+  /// the (wrong) way |oneStepAside| will contain any number of items.
+  void FindOneStepAsideRoadPoint(RoadPoint const & center, Joint::Id centerId,
+                                 vector<JointEdge> const & edges,
+                                 vector<Joint::Id> & oneStepAside) const;
+
+  bool GetIngoingAndOutgoingEdges(Joint::Id centerId, bool graphWithoutRestrictions,
```

Comment:
Add a comment to the method, please. It doesn't clear for me why it return false if at least number of outgoingEdges is empty. I mean, why it returns bool at all?